### PR TITLE
Task/api 423 appointment implementation

### PR DESCRIPTION
**Pull request by bschofield-va**

### CODE REVIEW

File: sentinel/src/test/java/gov/va/health/api/sentinel/ArgonautReadAndSearchIT.java [hunk 1]
```diff
@@ -67,6 +68,14 @@
...
             "/api/AllergyIntolerance?patient={patient}",
             ids.patient()),
+        // Appointment
+        // Allergy Intolerance
+        expect(200, Appointment.class, "/api/Appointment/{id}", ids.appointment()),
+        expect(404, OperationOutcome.class, "/api/Appointment/{id}", ids.unknown()),
+        expect(200, Appointment.Bundle.class, "/api/Appointment?_id={id}", ids.appointment()),
+        expect(
+            200, Appointment.Bundle.class, "/api/Appointment?identifier={id}", ids.appointment()),
+        expect(200, Appointment.Bundle.class, "/api/Appointment?patient={patient}", ids.patient()),
```
Comment (esubocz-va) [inline sentinel/src/test/java/gov/va/health/api/sentinel/ArgonautReadAndSearchIT.java]: search by patient?

File: argonaut/src/main/java/gov/va/api/health/argonaut/service/controller/appointment/AppointmentController.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      @RequestParam("identifier") String id,
+      @RequestParam(value = "page", defaultValue = "1") int page,
+      @RequestParam(value = "_count", defaultValue = "1") int count) {
+    return bundle(
+        Parameters.builder().add("identifier", id).add("page", page).add("_count", count).build(),
+        page,
+        count);
+  }
+
+  /** Search by patient. */
```
Comment (esubocz-va) [inline argonaut/src/main/java/gov/va/api/health/argonaut/service/controller/appointment/AppointmentController.java]: Are we supporting a search by patient for Appointment? Thought this resource would only include search by _id and search by identifier?

File: argonaut/src/test/java/gov/va/api/health/argonaut/service/controller/appointment/AppointmentControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    root.setPageNumber(BigInteger.valueOf(1));
+    root.setRecordsPerPage(BigInteger.valueOf(10));
+    root.setRecordCount(BigInteger.valueOf(3));
+    root.setAppointments(new CdwAppointments());
+    CdwAppointment cdwItem1 = new CdwAppointment();
+    CdwAppointment cdwItem2 = new CdwAppointment();
+    CdwAppointment cdwItem3 = new CdwAppointment();
+    root.getAppointments().getAppointment().addAll(Arrays.asList(cdwItem1, cdwItem2, cdwItem3));
+    Appointment patient1 = Appointment.builder().build();
+    Appointment patient2 = Appointment.builder().build();
```
Comment (eclendenning-va) [inline argonaut/src/test/java/gov/va/api/health/argonaut/service/controller/appointment/AppointmentControllerTest.java]: not patients

File: sentinel/src/test/java/gov/va/health/api/sentinel/ArgonautReadAndSearchIT.java [hunk 4]
```diff
@@ -67,6 +68,14 @@
             AllergyIntolerance.Bundle.class,
             "/api/AllergyIntolerance?patient={patient}",
             ids.patient()),
+        // Appointment
+        // Allergy Intolerance
```
Comment (eclendenning-va) [inline sentinel/src/test/java/gov/va/health/api/sentinel/ArgonautReadAndSearchIT.java]: remove allergy intolerance comment

File: argonaut/src/main/java/gov/va/api/health/argonaut/service/controller/appointment/AppointmentController.java (hunk 2 above)
Comment (bschofield-va) [reply argonaut/src/main/java/gov/va/api/health/argonaut/service/controller/appointment/AppointmentController.java]: The database supports searching by patient and WellHive will need this functionality, so I went ahead and added it

File: sentinel/src/test/java/gov/va/health/api/sentinel/ArgonautReadAndSearchIT.java (hunk 1 above)
Comment (bschofield-va) [reply sentinel/src/test/java/gov/va/health/api/sentinel/ArgonautReadAndSearchIT.java]: yup

**Comment by bschofield-va** [comment]
:facepalm:
I've shamed myself with that dead code :(

**Pull request merged**